Northstar.Client/mod/scripts/vscripts/ui/_menus.nut

```diff
@@ -178,6 +178,20 @@ void function UICodeCallback_ToggleInGameMenu()
 	}
 }
 
+void function CancelLevelLoadingInfinite()
+{
+	//Singleplayer "Continue" counts as finished loading and shouldn't be affected
+	print("CancelLevelLoadingInfinite has Initialized")
+	wait 30
+	if (uiGlobal.isLoading == true)
+	{
+		printt("CancelLevelLoadingInfinite disconnected because loading took too long")
+		UICodeCallback_LevelLoadingFinished( true )
+		Disconnect()
```

Comment:
Would definitely be nice to show the user a dialog here that explains things (tbh the "Connection Timed Out" dialog in vanilla would work)